app/src/androidTest/java/com/example/bigowlapp/activity/SignUpPageActivityTest.java

```diff
@@ -0,0 +1,140 @@
+package com.example.bigowlapp.activity;
+
+import android.os.SystemClock;
+import android.view.View;
+
+import androidx.lifecycle.Lifecycle;
+import androidx.test.espresso.Espresso;
+import androidx.test.espresso.UiController;
+import androidx.test.espresso.ViewAction;
+import androidx.test.espresso.matcher.ViewMatchers;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import com.example.bigowlapp.R;
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.viewModel.EditProfileViewModel;
+import com.example.bigowlapp.viewModel.SignUpViewModel;
+
+import org.hamcrest.Matcher;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.Random;
+
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.matcher.ViewMatchers.hasErrorText;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.isRoot;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static org.mockito.Mockito.atMostOnce;
+import static org.mockito.Mockito.verify;
+
+@RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+@LargeTest
+public class SignUpPageActivityTest {
+
+    @Rule
+    public ActivityScenarioRule<SignUpPageActivity> activityRule = new ActivityScenarioRule<>(SignUpPageActivity.class);
+
+    @Mock
+    private SignUpViewModel mockSignUpViewModel;
+
+    private String firstName = "John";
+    private String lastName = "Doe";
+    private String email = "john.doe@email.com";
+    private String password = "abc123";
+    private String phone;
+
+    SignUpPageActivity signUpPageActivity;
+
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void A_unSufficientSignUpInputTest() {
```

Comment:
I personally don't like this format of the method name, I think the descriptive names are enough and the letters at the begiinning don't seem to add any value. You can argue why you think it's valid though. Also starts with capital letter which isn't java standared for methods.

Reply:
I know what you mean, but as I mentioned above, it for the test be test in alphabetical order.

Reply:
I removed it.